Assets/Scripts/InventoryAndItemsScripts/AffectPlayerHealthFromArea.cs

```diff
@@ -33,19 +37,36 @@ void OnDisable()
 
         void IncreaseHealth(GameObject player)
         {
-            if (this.healthIncreaseWhenPass > 0)
+            if (this.healthIncreaseWhenPass > 0 && !wait)
             {
-                //Would replace with call to player and update player health
+                wait = true;
+                player.GetComponent<PlayerManager>().IncreaseHealth(healthIncreaseWhenPass);
                 Debug.Log("Yay! Health +" + this.healthIncreaseWhenPass);
+            } else
+            {
+                elapsed -= Time.deltaTime;
             }
         }
 
         void DecreaseHealth(GameObject player)
         {
-            if (this.healthDecreaseWhenFail > 0)
+            if (this.healthDecreaseWhenFail > 0 && !wait)
             {
-                //Would replace with call to player and update player health
+                wait = true;
+                player.GetComponent<PlayerManager>().DecreaseHealth(healthDecreaseWhenFail);
                 Debug.Log("Ow! Health -" + this.healthDecreaseWhenFail);
+            } else
+            {
+                elapsed -= Time.deltaTime;
```

Comment:
this should probably just be decremented in the update loop, i'm  sure you're calling it in some update loop, but having it be explicitly in an update loop is better for code quality

Reply:
This totally makes sense, I'm actually not sure why it wasn't doing this already...I feel like I tried to do it and there was a bug or something but I put it back now and it works fine so